Makefile

```diff
@@ -31,7 +31,11 @@ $(TOOLS_DIR)/stringer: go.mod go.sum tools.go
 
 precommit: $(TOOLS_DIR)/golangci-lint  $(TOOLS_DIR)/misspell $(TOOLS_DIR)/stringer
 	PATH="$(abspath $(TOOLS_DIR)):$${PATH}" go generate ./...
-	$(TOOLS_DIR)/golangci-lint run --fix # TODO: Fix this on windows.
+	# TODO: Fix this on windows.
+	for dir in $(ALL_GO_MOD_DIRS); do \
+	  echo "golangci-lint in $${dir}"; \
+	  (cd "$${dir}" && $(abspath $(TOOLS_DIR))/golangci-lint run --fix); \
+	done
```

Comment:
on Mac this doesn't run because  Line 11 
```
ALL_GO_MOD_DIRS := $(shell find . -name 'go.mod' -printf "%h\n" | sort)
```
fails with an error
```
find: -printf: unknown primary or operator
```

Reply:
This may be because I don't have GNU tools installed on my mac.

Reply:
What is "%h" anyway?

Reply:
From GNU find manual:
```
%h Leading directories of file's name (all but the last element). If the file name contains no slashes (since it is in the current directory) the %h specifier expands to ".".
```